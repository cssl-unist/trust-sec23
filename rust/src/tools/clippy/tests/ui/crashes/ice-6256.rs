// originally from rustc ./src/test/ui/regions/issue-78262.rs
// ICE: to get the signature of a closure, use substs.as_closure().sig() not fn_sig()

trait TT {}

impl dyn TT {
    fn func(&self) {}
}

fn main() {
    let f = |x: &dyn TT| x.func(); //[default]~ ERROR: mismatched types
                                   //[nll]~^ ERROR: borrowed data escapes outside of closure
}
