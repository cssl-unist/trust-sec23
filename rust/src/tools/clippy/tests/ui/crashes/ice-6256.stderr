error[E0308]: mismatched types
  --> $DIR/ice-6256.rs:11:28
   |
LL |     let f = |x: &dyn TT| x.func(); //[default]~ ERROR: mismatched types
   |                            ^^^^ lifetime mismatch
   |
   = note: expected reference `&(dyn TT + 'static)`
              found reference `&dyn TT`
note: the anonymous lifetime #1 defined on the body at 11:13...
  --> $DIR/ice-6256.rs:11:13
   |
LL |     let f = |x: &dyn TT| x.func(); //[default]~ ERROR: mismatched types
   |             ^^^^^^^^^^^^^^^^^^^^^
   = note: ...does not necessarily outlive the static lifetime

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
